src/objects.c

```diff
@@ -1604,7 +1605,51 @@ s32 func_8001C48C(s32 arg0) {
     return -1;
 }
 
-GLOBAL_ASM("asm/non_matchings/objects/func_8001C524.s")
+s32 func_8001C524(f32 xDiff, f32 yDiff, f32 zDiff, s32 someFlag) {
+    f32 pad[6];
+    s32 sp64;
+    f32 x;
+    f32 len;
+    f32 z;
+    f32 y;
+    f32 dist;
+    s32 var_a0;
+    s32 numSteps;
+    s32 result;
+    ObjectSegment* segment;
+    LevelObjectEntry_TTDoor* levelObj;
```

Comment:
```suggestion
    ObjectSegment *segment;
    LevelObjectEntry_TTDoor *levelObj;
```

Reply:
ah, ill make a note to try and remember to swap these, thanks for the fix